src/canmatrix/canmatrix.py

```diff
@@ -316,16 +319,16 @@ def phys2raw(self, value=None):
         raw_value = (value - self.offset) / self.factor
 
         if not self.is_float:
-            raw_value = int(raw_value)
-        return raw_value
+            raw_value = round(raw_value)
+        return float(raw_value)
```

Comment:
If we want to convert to a not-int then wouldn't a float factory be the way?  Specifically `self.float_factory` here it seems.  Though I'm thinking this is not needed as `self.offset` and `self.factor` should already be the proper types and as such `raw_value` would generally be the proper type already.

Reply:
Sorry, I see where you were going and yes, it makes sense.  Forget my comment.  Though, we are stuck without separate single and double float types.  I think I have gotten to see a couple float signals in the wild actually... Right before they got converted to fixed point.

Now that I think about it round doesn't actually give you an int, does it?

Reply:
```python
>>> type(round(4.999))
<class 'int'>
```
But I don't have python 2 to check it there.

Reply:
for me it's ok, if py3 works. 
Py2 is not primary development platform anymore. 
As I am not sure phys2raw will stay, because something is wrong with decoding/encoding anyway. So maybe API for decoding/encoding changes in future.

Reply:
I'd say we fully support py2 until we drop it entirely.  Unless we are going to pull a maintenance-only py2 branch, but even then the py2 branch should be fully py2 correct.  I don't think either is happening yet so let's not muddy the waters.  And yes, in py2 `round()` returns a float.  Easy enough to deal with in this case.

https://docs.python.org/2/library/functions.html#round
> The result is a floating point number.

Reply:
I'm still hesitant about the float thing.  It loses information (the exact value).  Of course, what's proper totally depends on the point of this function.  Is it to give you a type whose representation is exactly what would be on the bus?  Or to just handle the scaling?  Should it even be doing rounding vs. truncation?  So I guess I'm not sure what it should be but I'd like it consistent.  If we're going to float (which still misses single vs double precision) then it seems we ought to be returning an int for integer signals.  We can leave the bigger picture debates for later.